x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -285,6 +286,25 @@ public LogicalPlan visitFromCommand(EsqlBaseParser.FromCommandContext ctx) {
         );
     }
 
+    @Override
+    public PlanFactory visitInsistCommand(EsqlBaseParser.InsistCommandContext ctx) {
+        var source = source(ctx);
+        List<NamedExpression> fields = visitQualifiedNamePatterns(ctx.qualifiedNamePatterns(), ne -> {
+            Source neSource = ne.source();
+            if (ne instanceof UnresolvedStar) {
+                throw new ParsingException(neSource, "Cannot specify [*] with INSIST", neSource.text());
+            }
+            if (ne instanceof UnresolvedNamePattern) {
+                throw new ParsingException(neSource, "Cannot use wildcards ([*]) with INSIST", neSource.text());
+            }
+        });
+        return input -> new Insist(
+            source,
+            fields.stream().map(ne -> (Attribute) new UnresolvedAttribute(ne.source(), ne.name())).toList(),
```

Comment:
new ArrayList<Attribute>(fields) or change Insist to accept a List<? extends Attribute> (Similar to Project).